stage1/prepare-app/prepare-app.c

```diff
@@ -138,6 +139,40 @@ static void mount_at(const char *root, const mount_point *mnt)
 		 "Mounting \"%s\" on \"%s\" failed", mnt->source, to);
 }
 
+static int mount_sys_required(const char *root)
+{
+	FILE *f;
+	char *line = NULL;
+	size_t len = 0;
+	ssize_t read;
+
+	pexit_if((f = fopen("/proc/self/mountinfo", "re")) == NULL,
+		 "Unable to open /proc/self/mountinfo");
+
+	while ((read = getline(&line, &len, f)) != -1) {
+		char *sys_dir;
+		char *sys_subdir;
+		char *mountpoint;
+
+		asprintf(&sys_dir, "%s/sys", root);
+		asprintf(&sys_subdir, "%s/sys/", root);
```

Comment:
With `asprintf()`, I don't have to decide the buffer size though, so it seems better.

I should check the return value though.